perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/MembersManagerMethod.java

```diff
@@ -1742,7 +1744,20 @@ public Member call(ApiCaller ac, Deserializer parms) throws PerunException {
 			parms.stateChangingCheck();
 
 			Status status = Status.valueOf(parms.readString("status"));
-			return ac.getMembersManager().setStatus(ac.getSession(), ac.getMemberById(parms.readInt("member")), status);
+			PerunSession sess = ac.getSession();
+			Member member = ac.getMemberById(parms.readInt("member"));
+			// Set new status
+			member = ac.getMembersManager().setStatus(sess, member, status);
+			// Reject all applications for this member if member is expired
+			if (status != Status.EXPIRED) return member;
+			List<Group> groups = sess.getPerun().getGroupsManager().getMemberGroups(sess, member);
+			for (Group g : groups) {
+				List<Application> apps = ac.getRegistrarManager().getApplicationsForMember(sess, g, member);
+				for (Application app: apps) {
+					ac.getRegistrarManager().rejectApplication(sess, app.getId(), "Application is set to status REJECTED, because the member it is assigned to is in EXPIRED state.");
+				}
+			}
```

Comment:
This will also pass approved and already rejected applications into `rejectApplication()`, which will throw an exception. We only want to reject open applications. Also, maybe we can extract this into a separate method in RegistrarManager, to save some code (and include exception handling there)?